src/main/java/org/myrobotlab/codec/CodecUtils.java

```diff
@@ -270,20 +275,50 @@ public static String capitalize(final String line) {
      * using the selected JSON backend.
      *
      * @param json  The JSON to be deserialized in String form
-     * @param clazz The target class.
+     * @param clazz The target class. If a class is not supplied the default class returned will be an LinkedHashMap
      * @param <T>   The type of the target class.
      * @return An object of the specified class (or a subclass of) with the state
      * given by the json. Null is an allowed return object.
      * @throws JsonDeserializationException if an error during deserialization occurs.
      * @see #USING_GSON
      */
+    @SuppressWarnings("unchecked")
     public static <T> /*@Nullable*/ T fromJson(/*@Nonnull*/ String json, /*@Nonnull*/ Class<T> clazz) {
         try {
             if (USING_GSON) {
+                if (clazz == null) {
+                  clazz = (Class<T>)LinkedHashMap.class;
+                }
                 return gson.fromJson(json, clazz);
+            } else {
+              if (clazz == null) {
+ 
+                JsonFactory factory = new JsonFactory();
+                JsonParser parser = factory.createParser(json);
+
+                // "peek" at the next token to determine its type
+                JsonToken token = parser.nextToken();
+
+                if (token == JsonToken.START_OBJECT) {
+                  clazz = (Class<T>)Map.class;
+                } else if (token == JsonToken.START_ARRAY) {
+                  clazz = (Class<T>)ArrayList.class;
+                } else if (token.isScalarValue()) {
+                  ObjectMapper objectMapper = new ObjectMapper();

```

Comment:
This both bypasses the static mapper that may have important configuration details and creates a new mapper on every invocation, which is an expensive operation

Reply:
good call - will be using the static mapper - i assume its thread safe ... must be without exploding at this point

Reply:
It is thread safe. However, it might be possible to cause some performance issues in a heavily-threaded environment due to the internal locking... Might be a good idea to transition the single static field to a thread local at some point, so the mapper is only created once per thread but no performance issues due to locking are caused. Not a concern right now, but it's definitely an optimization we can make later

Reply:
good insight ... pr for later